extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/CachedChannelProvider.java

```diff
@@ -45,6 +61,14 @@ enum ChannelType {
     private final RAPriQueue<PerPathPool> releasePriority =
             new RAPriQueue<>(8, PerPathPool.RAPQ_ADAPTER, PerPathPool.class);
 
+    /**
+     * Stores all channels (not just the pooled ones) created by this provider for all paths. Used for invalidating file
+     * handles associated with this provider.
+     */
+    private final Collection<WeakReference<SeekableByteChannel>> channelList = new ArrayList<>();
```

Comment:
Having this use weak refs is probably unnecessary. In general, it's an error (resource leak) to make a channel and not close it. The code we have using these channels is very careful about that.

Maybe we want to:
1. Integrate "outstanding channel tracking" into `PerPathPool`.
2. Make `CachedChannel` instances aware of their pool.
3. The pool can then be aware of available and outstanding channels, and handle invalidating both (and itself).

I'm trying to think of a way to get rid of more of the data structures here, but not at a useful point yet.